share/wake/lib/system/job.wake

```diff
@@ -97,7 +97,7 @@ tuple Plan =
     # The label used when showing the command during execution. This is recommended for efficient
     # debugging and locating job information. For example, the label will show up in the terminal during
     # job execution, in a job's progress bar in the terminal, when running the `wake --last` command,
-    # and in the `<..>.in.json` and `<...>.out.json` files in the `.build` directory.
+    # and in the `<..>.spec` and `<...>.result` files in the `.build` directory.
```

Comment:
Can we keep the ".json" part? It's useful for syntax highlighting and the like.

Reply:
I'd actually rather have no extension, but wanted to separate out the spec and result. I see people putting the encoding type as file extensions a lot, but it's just implementation detail. Syntax highlighting for json specifically is pretty low value compared to having it pretty-printed.

Perhaps we could swap the name order to become `spec-{jobid}` and `result-{jobid}`

Reply:
I'm fine with no json extension. Does that give us some extra value or is it just a preference?

Reply:
It's a preference, as a lot of people end up call things in discussions the $thing json rather than $thing

Reply:
+1 on dropping the extension.  It's unfortunate that programs/platforms treat that as the sole marker, but especially since we don't expect the average user to be editing these, we shouldn't be constrained by implementation details.

Reply:
No, the assumption is correct: "what a file's format is **intended but not guaranteed** to be".  If it makes sense to use a standard file extension in some situation, then there's no need to be going off looking for something unique.  For things like JSON, though, where there's a (maybe implicit) schema *in addition to* the syntax, and especially in this case where we have two different schemas being used in the same directory, it sometimes makes more sense to treat that as the primary thing to be naming files after rather than the agnostic format.  (On the other hand, if you're using a file as interchange between different programs with a looser/no schema, then you definitely *should* be using the format extension.)

Reply:
> What is the point of file extensions?

On unix-like systems, there is no inherent purpose/mechanism other than visually communicating to the user. It's just bytes.

It's expected that if something really does need to know file-type, it should inspect it via something like `file`.
On a graphical system, some use programs like `xdg-open` to guess associated applications for use with file extensions like on windows

Reply:
@bmitc what does vscode do if there are no extensions? I seem to recall it assuming that some of my log files without extensions were yaml files for some reason.

Reply:
Also I would still like to say, I mostly consider this a bike shed but my preference for the color of the bike shed at the nuclear facility is "with extension"